physics/dynamic_frame_test.cpp

```diff
@@ -400,6 +401,103 @@ TEST_F(DynamicFrameTest, CentrifugalAcceleration) {
                            0));
 }
 
+// A frame initially nonrotating and in uniformly accelerated rotation around
+// the origin.  The test point is on the x axis.  The acceleration is purely due
+// to Euler.  The motion elements that don't have specific values have no effect
+// on the acceleration.
+TEST_F(DynamicFrameTest, EulerAcceleration) {
+  Instant const t0;
+
+  EXPECT_CALL(mock_frame_, MotionOfThisFrame(_))
+      .WillRepeatedly(Invoke([t0](Instant const& t) {
+        AngularAcceleration const ωʹ = 10 * Radian / Second / Second;
+        Bivector<AngularAcceleration, Inertial> const
+            angular_acceleration_of_to_frame({0 * Radian / Second / Second,
+                                              0 * Radian / Second / Second,
+                                              ωʹ});
+        Rotation<Inertial, Rotating> const rotation(
+            ωʹ * Pow<2>(t - t0) / 2,
+            angular_acceleration_of_to_frame,
+            DefinesFrame<Rotating>{});
+        RigidTransformation<Inertial, Rotating> const
+            rigid_transformation(
+                /*from_origin=*/Inertial::origin,
+                /*to_origin=*/Rotating::origin,
+                rotation.Forget<OrthogonalMap>());
+        AngularVelocity<Inertial> const angular_velocity_of_to_frame;
+        Velocity<Inertial> const velocity_of_to_frame_origin;
+        RigidMotion<Inertial, Rotating> const rigid_motion(
+            rigid_transformation,
+            angular_velocity_of_to_frame,
+            velocity_of_to_frame_origin);
+        Vector<Acceleration, Inertial> const
+            acceleration_of_to_frame_origin;
+        return AcceleratedRigidMotion<Inertial, Rotating>(
+            rigid_motion,
+            angular_acceleration_of_to_frame,
+            acceleration_of_to_frame_origin);
+      }));
+
+  // No gravity.
+  Vector<Acceleration, Inertial> const gravitational_acceleration;
+  EXPECT_CALL(mock_frame_, GravitationalAcceleration(_, _))
+      .WillRepeatedly(Return(gravitational_acceleration));
+
+  // The test point is on the x axis.
+  DegreesOfFreedom<Rotating> const initial_state_in_rotating_frame = {
+      Rotating::origin + Displacement<Rotating>({100 * Metre,
+                                                 0 * Metre,
+                                                 0 * Metre}),
```

Comment:
While our APIs make it very hard to treat x and y differently at this level, z is another story, since it is the axis. In this test and the ones above, use a nonzero speed and displacement in z (preferably with numbers that looks nothing like the ones we are going to encounter, _e.g._, 17 * Metre).